searchlib/src/vespa/searchlib/attribute/multinumericpostattribute.hpp

```diff
@@ -145,13 +145,18 @@ MultiValueNumericPostingAttribute<B, M>::DocumentWeightAttributeAdapter::make_bi
     return self.getPostingList().make_bitvector_iterator(idx, doc_id_limit, match_data, strict);
 }
 
+template <typename B, typename M>
+bool
+MultiValueNumericPostingAttribute<B, M>::DocumentWeightAttributeAdapter::has_weight_iterator(vespalib::datastore::EntryRef idx) const noexcept
+{
+    return self.getPostingList().has_btree(idx);
+}
+
 template <typename B, typename M>
 const IDocumentWeightAttribute *
 MultiValueNumericPostingAttribute<B, M>::asDocumentWeightAttribute() const
 {
-    if (this->hasWeightedSetType() &&
-        this->getBasicType() == AttributeVector::BasicType::INT64 &&
-        !this->getConfig().getIsFilter()) {
+    if (this->hasWeightedSetType() && (this->getBasicType() == AttributeVector::BasicType::INT64)) {
```

Comment:
The `create()` member functions in `MultiValueNumericPostingAttribute` and `MultiValueStringPostingAttributeT` only operate with `DocumentWeightIterator` instances. This means that a posting list that only has a bitvector representation and not a btree representation will incorrectly be considered as an empty posting list. This is likely the cause for the observed regression.